### ISSUE TITLE
Not works with AnimatedList inside

### ISSUE THREAD

**Issue by vanosidor** [peng8350/flutter_pulltorefresh#108]
With ListView in child  of SmartRefresher it works perfectly. If I try to change ListView with AnimatedList I have empty screen with no items. My SmartRefresher widget: 
```
          SmartRefresher(
            footer: ClassicFooter(
              idleText: Texts.to_load_more_tasks,
              noDataText: Texts.no_more_tasks,
              loadingText: Texts.wait_loading,
              onClick: () {
                if (_refreshController.footerStatus != LoadStatus.idle) return;
                _onLoading();
              },
            ),
            enablePullDown: true,
            enablePullUp: true,
            header: WaterDropMaterialHeader(
              backgroundColor: widget.tabColor,
            ),
            controller: _refreshController,
            onRefresh: _onRefresh,
            onLoading: _onLoading,
            child: AnimatedList(
              shrinkWrap: true,
              initialItemCount: _taskList.length,
              itemBuilder: (BuildContext context, int index, Animation animation) =>
                  _buildElement(index),
            ),
          ),
```
![Снимок экрана от 2019-06-13 20-51-49](https://user-images.githubusercontent.com/14176528/59458643-8cdffd80-8e23-11e9-8fe5-1d873f71124f.png)

If I use ListView I have this:

```
    child: ListView.builder(
      itemCount: _taskList.length,
      itemBuilder: (BuildContext context, int index) =>
          _buildElement(index),
    ),
```

![Снимок экрана от 2019-06-13 21-51-52](https://user-images.githubusercontent.com/14176528/59459440-7c308700-8e25-11e9-9e4a-daa045511ac5.png)

**Comment by peng8350** [comment]
there are two ways to solve this:
1.copy code from animated_list.dart,provide provide SmartRefresher's parameter to AnimatedList,in it's state build method:

```
  @override
  Widget build(BuildContext context) {
    return SmartRefresher(
      child: ListView.builder(
        itemBuilder: _itemBuilder,
        itemCount: _itemsCount,
        scrollDirection: widget.scrollDirection,
        reverse: widget.reverse,
        controller: widget.controller, 
        primary: widget.primary,
        physics: widget.physics,
        shrinkWrap: widget.shrinkWrap,
        padding: widget.padding,
      ),
      controller: widget.controller,
      enablePullUp: widget.enablePullUp,
      ....
    );
  }

```

2.Add this code in your code that you provided
```
 physics: NeverScrollableScrollPhysics(),
```

Suggest use the first way,When the amount of data is large,second way may have a performance problem.On the other hand, it's not that I don't want to be compatible with such widgets (SingleChildScrollView, AnimatedList), but that I don't know how to get internal slivers.they are not extends ScrollView,but build method return inner Scrollable widget

**Comment by vanosidor** [comment]
Thanks for response! The 2nd method does not work, the same result. Can you explain more detail 1st method?

**Comment by peng8350** [comment]
I have added a new commit that can solve your problem,[here](https://github.com/peng8350/flutter_pulltorefresh/blob/master/example/lib/other/refresh_animatedlist.dart),copy it and use,[here](https://github.com/peng8350/flutter_pulltorefresh/blob/master/example/lib/ui/test/Example4.dart) is example

**Comment by peng8350** [comment]
may be this problem can be solved after flutter adding SliverAnimatedList,but it has been reverted.I don't know why

**Comment by vanosidor** [comment]
Thanks for the great solution! It works fine for me!

**Comment by peng8350** [comment]
I reopen it,it indeed a problem in my plugin,If ScrollView wrap in another widget,It will not work.

**Comment by peng8350** [comment]
Flitter has merged the sliceranimatedlist into the code,you should update the new version flutter, and use CustomScrollView+SliverAnimatedList

**Comment by ovideozn** [comment]
Can you please provide an example of how SliverAnimatedList should be used? It is now available in the latest stable Flutter version 1.12.13, but I still can't get it to work with pull-to-refresh.
